api/tests/unit/app_analytics/test_tasks.py

```diff
@@ -0,0 +1,71 @@
+from datetime import datetime
+
+import pytest
+from app_analytics.models import APIUsageBucket, APIUsageRaw, Resource
+from app_analytics.tasks import populate_bucket
+from django.utils import timezone
+
+
+@pytest.mark.django_db(databases=["analytics"])
+def test_populate_bucket_15m_bucket():
+    # Given
+    environment_id = 1
+    bucket_size = 15
+    now = timezone.now()
+
+    for _ in range(5):
+        create_events(environment_id, 2, now)
+        now = now - timezone.timedelta(minutes=15)
+
+    # When
+    populate_bucket(bucket_size, process_last=50)
+
+    # Then
+    assert APIUsageBucket.objects.filter(bucket_size=15).count() == 3
+
+    # all three buckets are 15 minutes apart
+    buckets = APIUsageBucket.objects.all().order_by("created_at")
+
+    buckets[0].created_at == buckets[1].created_at + timezone.timedelta(
+        minutes=15
+    ) == buckets[0].created_at + timezone.timedelta(minutes=30)
```

Comment:
I think this should be `buckets[2].created_at + ...` ?

Reply:
Yep